src/jsCore.ts

```diff
@@ -0,0 +1,509 @@
+/*
+ * Copyright (c) 2020-2022 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+
+'use strict';
+
+import type {
+  InitTrackerConfiguration,
+  SelfDescribing,
+  EventContext,
+  StructuredProps,
+  ScreenViewProps,
+  ScreenSize,
+  PageViewProps,
+  TimingProps,
+  ConsentGrantedProps,
+  ConsentWithdrawnProps,
+  EcommerceTransactionProps,
+  DeepLinkReceivedProps,
+  MessageNotificationProps,
+  NetworkConfiguration,
+} from './types';
+
+import {
+  Payload,
+  trackerCore,
+  PayloadBuilder,
+  buildSelfDescribingEvent,
+  buildStructEvent,
+  buildScreenView,
+  buildPageView,
+  buildConsentGranted,
+  buildConsentWithdrawn,
+  buildEcommerceTransaction,
+} from '@snowplow/tracker-core';
+import type { TrackerCore } from '@snowplow/tracker-core';
+import { errorHandler } from './utils';
+
+// Tracker version added to the events
+const trackerVersion = 'rn-1.3.0';
+
+interface Tracker extends TrackerCore {
+  setDomainUserId: (duid: string | undefined) => void;
+  setNetworkUserId: (nuid: string | undefined) => void;
+}
+
+let trackers: { [namespace: string]: Tracker } = {};
+
+function preparePayload(payload: Payload): Record<string, string> {
+  const stringifiedPayload: Record<string, string> = {};
+
+  payload['stm'] = new Date().getTime().toString();
+
+  for (const key in payload) {
+    if (Object.prototype.hasOwnProperty.call(payload, key)) {
+      stringifiedPayload[key] = String(payload[key]);
+    }
+  }
+  return stringifiedPayload;
+}
+
+function createEmitCallback(networkConfig: NetworkConfiguration): (e: Payload) => void {
+  return (e: Payload) => {
+    const postJson = {
+      schema:
+        'iglu:com.snowplowanalytics.snowplow/payload_data/jsonschema/1-0-4',
```

Comment:
nit - would it make sense to move the schema strings (for `payload_data`, `deep_link_received`, `message_notification`, `timing`) e.g. to `constants.ts`? Not an objection at all though, just for better visibility.

Reply:
Moved to constants